### DISCUSSION TITLE
Large datasets will only render 100 rows at any time

### DISCUSSION THREAD

**Discussion by cbroome** [dowjones/react-dropdown-tree-select#451] [General]
**Describe the bug**
I have a large data set I'm loading into the select tool at one time, there's about 20 parent nodes, but each parent can have thousands of children. When expanding such a parent only 100 rows will be displayed at any time, this includes parents and children. There seems to be no way to get the remainder of the rows to display properly. 

Looking at the code briefly it seems this is an issue with the infinite scroll plugin and not calculating the total number of nodes properly. 

**To Reproduce**
Please include steps to reproduce the behavior;

I set up an example here: 
https://codesandbox.io/s/using-bootstrap-styles-with-react-dropdown-tree-select-forked-9nkwb?file=/index.js

I used md5s to obscure the names but you'll notice that expanding the second parent node will remove the following parent nodes and only render a subset of the children.

**Expected behavior**
All child rows should be visible if not scrollable. Parent nodes should always be visible.

**Environment (please complete the following information):**
 - Chrome 87.0.4280.88 
 - OS: MacOS

**Comment by mrchief** [comment]
@cbroome That's by [design](https://github.com/dowjones/react-dropdown-tree-select/issues/80). As you scroll up or down, it'll render the remaining rows. Rendering every single row slows the page down. The user cannot see beyond 100 records in most cases anyway.

**Reply by cbroome** [reply]
@mrchief - Thanks for the quick response

> Rendering every single row slows the page down. 

I agree with that logic. 

> As you scroll up or down, it'll render the remaining rows

I'm not seeing that behavior with the example posted. Scrolling to the bottom of the displayed rows just stops. If there's a trigger to render the next 100 entries, it's not firing.

**Comment by mrchief** [comment]
I just checked the docs page:

![Ckjs78zbdV](https://user-images.githubusercontent.com/781818/101921326-e05b1500-3b9a-11eb-8cb4-7a18be783341.gif)

**Reply by cbroome** [reply]
Sorry I didn't clarify in my earlier comment. I am seeing this issue when there's no filter applied. So a treeview with many child nodes will only display 100 max and not refresh on scroll if there's no filter.

**Reply by mrchief** [reply]
Can you post a gif of where it's not working? I checked without filters as well.

**Reply by cbroome** [reply]
Here you go. 

![react-tree-select](https://user-images.githubusercontent.com/521476/101925967-86f5e480-3ba0-11eb-8991-2c91f151acea.gif)

There are many more children to display, and the parent nodes become inaccessible as well.

**Reply by mrchief** [reply]
Seems like you're missing a css to add overflow

**Reply by cbroome** [reply]
Looks like it's applied by default to the react-dropdown-tree-select element. Was there another element that needs overflow applied? 

![Screen Shot 2020-12-15 at 9 21 11 AM](https://user-images.githubusercontent.com/521476/102226959-03493a00-3eb7-11eb-95e5-03755a09016c.png)


https://codesandbox.io/s/using-bootstrap-styles-with-react-dropdown-tree-select-forked-9nkwb?file=/index.js

**Reply by mrchief** [reply]
No not that one, the [outer div](https://github.com/dowjones/react-dropdown-tree-select/blob/develop/docs/src/stories/Simple/index.css#L7). Are you bringing in the [default styles](https://github.com/dowjones/react-dropdown-tree-select#default-styles)?

**Reply by cbroome** [reply]
Ah that was it. Applying `max-height` and `overflow-y` to `dropdown-content` fixed the issue. Thanks for sticking with it.
